kroxylicious-schema-validation/src/main/java/io/kroxylicious/proxy/filter/schema/JsonValueValidator.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.kroxylicious.proxy.filter.schema;
+
+import java.nio.ByteBuffer;
+
+import org.apache.kafka.common.record.Record;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import io.kroxylicious.proxy.filter.schema.validation.RecordValidator;
+
+/**
+ * Checks if a Record's value is valid JSON by attempting to deserialize it
+ * with jackson
+ */
+public class JsonValueValidator implements RecordValidator {
+
+    ObjectMapper mapper;
+
+    private JsonValueValidator() {
+        mapper = new ObjectMapper();
+    }
+
+    @Override
+    public Result validate(Record record) {
+        ByteBuffer value = record.value();
+        if (value == null) {
+            // maybe this should be configurable? a null value is often used to tombstone data but maybe users might
+            // want to prevent null values going to their topic
+            return VALID;
+        }
+        value.mark();
+        try {
+            byte[] bytes = new byte[record.valueSize()];
+            value.get(bytes);
+            mapper.readTree(bytes);
+            return VALID;
+        }
+        catch (Exception e) {
+            String message = "value was not valid JSON" + (e.getMessage() != null ? ": " + e.getMessage() : "");
```

Comment:
I wonder if we should use "well-formed" rather than "valid"? To me well-formed doesn't imply the possible existence of a schema, like "valid" does.

Reply:
I renamed to `SyntacticallyCorrect`, "well-formed" works for me too though